File: app/src/main/java/me/imzack/app/ender/view/contract/TypeCreationViewContract.java
package me.imzack.app.ender.view.contract;

import android.support.annotation.DrawableRes;

import me.imzack.app.ender.model.bean.FormattedType;

public interface TypeCreationViewContract extends BaseViewContract {

    void showInitialView(FormattedType formattedType);

    void onTypeNameChanged(String typeName, String firstChar);

    void onTypeMarkColorChanged(int colorInt, String colorName);

    void onTypeMarkPatternChanged(boolean hasPattern, @DrawableRes int patternResId, String patternName);

    void showTypeNameEditorDialog(String defaultName);

    void showTypeMarkColorPickerDialog(String defaultColor);

    void showTypeMarkPatternPickerDialog(String defaultPattern);
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/TypeCreationViewContract.kt
package net.zackzhang.app.end.view.contract

import android.support.annotation.DrawableRes

import net.zackzhang.app.end.model.bean.FormattedType

interface TypeCreationViewContract : BaseViewContract {

    fun showInitialView(formattedType: FormattedType)

    fun onTypeNameChanged(typeName: String, firstChar: String)

    fun onTypeMarkColorChanged(colorInt: Int, colorName: String)

    fun onTypeMarkPatternChanged(hasPattern: Boolean, @DrawableRes patternResId: Int, patternName: String?)

    fun showTypeNameEditorDialog(defaultName: String)

    fun showTypeMarkColorPickerDialog(defaultColor: Int)

    fun showTypeMarkPatternPickerDialog(defaultPattern: String?)
}
